### DISCUSSION TITLE
musl vs gnu

### DISCUSSION THREAD

**Discussion by lamyergeier** [dathere/qsv#1477] [Q&A] [answered]
I am using Fedora 39 on x86_64 architecture, so I have to install one of the following files (but which one?)
- qsv-0.117.0-x86_64-unknown-linux-gnu.zip
- qsv-0.117.0-x86_64-unknown-linux-musl.zip 

What is musl?

**Comment by jqnatividad** [comment, marked as answer]
gnu is not fully statically linked. It still depends on the right version of the gnu libraries being installed on your system.

musl is FULLY statically linked and self-contained with no external dependencies. Find out more at http://musl.libc.org/about.html

Insofar as performance is concerned, I haven't ran any benchmarks, but you can easily tweak the benchmark suite and find out yourself how it runs on Fedora 39 and your hardware.

**Comment by lamyergeier** [comment]
If we see the following: musc version is smaller than that of gnu:
qsv-0.119.0-x86_64-unknown-linux-gnu.zip  73.8 MB
qsv-0.119.0-x86_64-unknown-linux-musl.zip 53.3 MB

> musl is FULLY statically linked and self-contained with no external dependencies. 

Why the size of musl version smaller than that of gnu?  If its self contained, shouldn't it be bigger in size? Also should musl better than gnu as it is smaller, self contained with no external dependencies?

**Comment by jqnatividad** [comment]
The musl zip archive is smaller as not all features are enabled in the musl build as some crates fail to compile using musl.

That's why its smaller.
